### PR TITLE
NVTX hipification

### PR DESCRIPTION
**Pull request by shbiswas834**

### CODE REVIEW

**Review by lcskrishna — APPROVED** [review]
LGTM. @shbiswas834  Please add the full test run log here and we can merge this PR.

**Comment by shbiswas834** [comment]
@lcskrishna 
---------------------- TEST SUMMARY ------------------
tests/cupy_tests/array_api_tests|0| 8 skipped in 0.09s 
tests/cupy_tests/binary_tests|15| 15 passed in 3.19s 
tests/cupy_tests/core_tests|2872| short test summary info 
tests/cupy_tests/creation_tests|506| 2 failed, 406 passed, 93 skipped, 5 deselected in 25.04s 
tests/cupy_tests/cuda_tests|268| 9 failed, 158 passed, 96 skipped, 6 deselected in 7.69s 
tests/cupy_tests/fft_tests|2711| 437 failed, 1397 passed, 876 skipped, 1 deselected in 370.53s (0:06:10) 
tests/cupy_tests/functional_tests|73| 73 passed, 1 warning in 179.66s (0:02:59) 
tests/cupy_tests/indexing_tests|406| 405 passed, 1 deselected in 9.01s 
tests/cupy_tests/io_tests|42| 42 passed in 2.84s 
tests/cupy_tests/lib_tests|576| 123 failed, 411 passed, 42 xfailed, 12 warnings in 209.52s (0:03:29) 
tests/cupy_tests/linalg_tests|1395| 231 failed, 1148 passed, 16 skipped, 28 warnings in 177.80s (0:02:57) 
tests/cupy_tests/logic_tests|824| 824 passed, 2 warnings in 10.39s 
tests/cupy_tests/manipulation_tests|585| 2 failed, 547 passed, 32 skipped, 1 deselected, 3 xfailed in 31.09s 
tests/cupy_tests/math_tests|10192| short test summary info 
tests/cupy_tests/misc_tests|31| 31 passed in 14.40s 
tests/cupy_tests/padding_tests|75| 75 passed in 3.99s 
tests/cupy_tests/polynomial_tests|55| 1 failed, 54 passed in 11.24s 
tests/cupy_tests/prof_tests|8| 8 passed in 0.58s 
tests/cupy_tests/random_tests|1977| test session starts 
tests/cupy_tests/sorting_tests|499| 1 failed, 432 passed, 64 skipped, 1 deselected, 1 xfailed in 57.66s 
tests/cupy_tests/statistics_tests|733| 7 failed, 723 passed, 3 deselected, 83 warnings in 226.68s (0:03:46) 
tests/cupy_tests/test_cublas.py|2700| 2011 failed, 652 passed, 4 xfailed, 33 errors in 234.77s (0:03:54) 
tests/cupy_tests/testing_tests|293| 29 failed, 225 passed, 29 skipped, 10 xfailed in 8.03s 
tests/cupy_tests/test_init.py|18| 17 passed, 1 skipped in 4.77s 
tests/cupy_tests/test_ndim.py|12| 12 passed in 2.66s 
tests/cupy_tests/test_numpy_interop.py|11| 2 failed, 8 passed, 1 xfailed in 2.23s 
tests/cupy_tests/test_type_routines.py|34| 34 passed in 1.69s 
tests/cupy_tests/test_typing.py|1 item| 1 passed in 0.56s 
tests/cupyx_tests/distributed_tests|59| 43 failed, 16 passed in 136.33s (0:02:16) 
tests/cupyx_tests/fallback_mode_tests|107| 1 failed, 105 passed, 1 skipped in 4.88s 
tests/cupyx_tests/jit_tests|172| 135 passed, 19 skipped, 18 xfailed, 6 warnings in 9.09s 
tests/cupyx_tests/linalg_tests|24| 19 failed, 1 passed, 5 skipped in 4.59s 
tests/cupyx_tests/profiler_tests|19| 3 failed, 16 passed in 2.95s 
tests/cupyx_tests/scipy_tests/fftpack_tests|901| 901 passed in 164.32s (0:02:44) 
tests/cupyx_tests/scipy_tests/fft_tests|12237| 11657 passed, 420 skipped, 160 xpassed in 467.08s (0:07:47) 
tests/cupyx_tests/scipy_tests/interpolate_tests|592| 90 failed, 381 passed, 120 skipped, 1 deselected in 35.78s 
tests/cupyx_tests/scipy_tests/linalg_tests|468| test session starts 
tests/cupyx_tests/scipy_tests/ndimage_tests|33626| 848 passed, 32775 skipped, 3 deselected in 88.75s (0:01:28) 
tests/cupyx_tests/scipy_tests/signal_tests|9307| short test summary info 
tests/cupyx_tests/scipy_tests/sparse_tests|24205| short test summary info 
tests/cupyx_tests/scipy_tests/spatial_tests|536| 536 skipped in 2.32s 
tests/cupyx_tests/scipy_tests/special_tests|483| 474 passed, 9 skipped, 34 warnings in 8.71s 
tests/cupyx_tests/scipy_tests/stats_tests|201| 201 passed in 7.04s 
tests/cupyx_tests/scipy_tests/test_get_array_module.py|2| 2 passed in 3.07s 
tests/cupyx_tests/test_cudnn.py|1829| 1829 skipped in 4.01s 
tests/cupyx_tests/test_cupyx.py|2| 2 passed in 0.99s 
tests/cupyx_tests/test_cusolver.py|648| 32 failed, 616 skipped in 6.52s 
tests/cupyx_tests/test_cusparse.py|1326| 803 failed, 14 passed, 146 skipped, 363 xfailed in 226.70s (0:03:46) 
tests/cupyx_tests/test_cutensor.py|79| 79 skipped in 1.15s 
tests/cupyx_tests/test_lapack.py|70| 45 failed, 4 passed, 21 skipped in 8.21s 
tests/cupyx_tests/test_optimize.py|9| 9 skipped in 1.10s 
tests/cupyx_tests/test_pinned_array.py|76| 72 passed, 4 deselected in 4.19s 
tests/cupyx_tests/test_rsqrt.py|1 item| 1 passed in 3.05s 
tests/cupyx_tests/test_runtime.py|2| 1 passed, 1 skipped in 2.30s 
tests/cupyx_tests/test_time.py|9| 1 failed, 8 passed in 3.09s 
tests/cupyx_tests/tools_tests|21| 6 passed, 15 deselected in 6.53s 
tests/example_tests|12| 5 failed, 2 passed, 5 skipped in 13.76s 
tests/install_tests|12| 9 passed, 3 skipped in 1.39s

**Comment by lcskrishna** [comment]
@shbiswas834 Looks like there are a lot of failures. Make sure to verify if your changes caused any issues.

**Comment by shbiswas834** [comment]
https://amdcloud-my.sharepoint.com/:u:/g/personal/shbiswas_amd_com/EYhye5GKpX5Oq08SxOEy1U4BYQPWyawCZ6nDKKK-IhW8iA?e=jW1ZdD
Test suite logs in the rocm pytorch latest docker

---------------------- TEST SUMMARY ------------------
tests/cupy_tests/array_api_tests|0| 8 skipped in 0.60s 
tests/cupy_tests/binary_tests|15| 15 passed in 2.71s 
tests/cupy_tests/core_tests|2872| short test summary info 
tests/cupy_tests/creation_tests|506| 2 failed, 406 passed, 93 skipped, 5 deselected in 329.64s (0:05:29) 
tests/cupy_tests/cuda_tests|268| 9 failed, 158 passed, 96 skipped, 6 deselected in 7.62s 
tests/cupy_tests/fft_tests|2711| 437 failed, 1397 passed, 876 skipped, 1 deselected in 338.31s (0:05:38) 
tests/cupy_tests/functional_tests|73| 73 passed, 1 warning in 794.72s (0:13:14) 
tests/cupy_tests/indexing_tests|406| 405 passed, 1 deselected in 446.34s (0:07:26) 
tests/cupy_tests/io_tests|42| 42 passed in 2.70s 
tests/cupy_tests/lib_tests|576| 121 failed, 413 passed, 42 xfailed, 12 warnings in 251.57s (0:04:11) 
tests/cupy_tests/linalg_tests|1395| 228 failed, 1151 passed, 16 skipped, 29 warnings in 444.76s (0:07:24) 
tests/cupy_tests/logic_tests|824| 824 passed, 2 warnings in 280.41s (0:04:40) 
tests/cupy_tests/manipulation_tests|585| short test summary info 
tests/cupy_tests/math_tests|10192| test session starts 
tests/cupy_tests/misc_tests|31| 31 passed in 14.11s 
tests/cupy_tests/padding_tests|75| 75 passed in 14.91s 
tests/cupy_tests/polynomial_tests|55| 55 passed in 43.71s 
tests/cupy_tests/prof_tests|8| 8 passed in 1.01s 
tests/cupy_tests/random_tests|1977| short test summary info 
tests/cupy_tests/sorting_tests|499| short test summary info 
tests/cupy_tests/statistics_tests|733| 7 failed, 723 passed, 3 deselected, 83 warnings in 736.93s (0:12:16) 
tests/cupy_tests/test_cublas.py|2700| 2044 failed, 652 passed, 4 xfailed in 254.32s (0:04:14) 
tests/cupy_tests/testing_tests|293| 29 failed, 231 passed, 23 skipped, 10 xfailed in 20.67s 
tests/cupy_tests/test_init.py|18| 17 passed, 1 skipped in 4.91s 
tests/cupy_tests/test_ndim.py|12| 12 passed in 2.47s 
tests/cupy_tests/test_numpy_interop.py|11| 2 failed, 8 passed, 1 xfailed in 2.43s 
tests/cupy_tests/test_type_routines.py|34| 34 passed in 1.89s 
tests/cupy_tests/test_typing.py|1 item| 1 passed in 1.00s 
tests/cupyx_tests/distributed_tests|59| 43 failed, 16 passed in 116.87s (0:01:56) 
tests/cupyx_tests/fallback_mode_tests|107| 1 failed, 105 passed, 1 skipped in 8.39s 
tests/cupyx_tests/jit_tests|172| 135 passed, 19 skipped, 18 xfailed, 22 warnings in 732.25s (0:12:12) 
tests/cupyx_tests/linalg_tests|24| 20 failed, 5 skipped in 4.12s 
tests/cupyx_tests/profiler_tests|19| 3 failed, 16 passed in 2.97s 
tests/cupyx_tests/scipy_tests/fftpack_tests|901| 901 passed in 126.76s (0:02:06) 
tests/cupyx_tests/scipy_tests/fft_tests|12237| 11657 passed, 420 skipped, 160 xpassed in 463.66s (0:07:43) 
tests/cupyx_tests/scipy_tests/interpolate_tests|592| 223 failed, 248 passed, 120 skipped, 1 deselected in 85.84s (0:01:25) 
tests/cupyx_tests/scipy_tests/linalg_tests|468| test session starts 
tests/cupyx_tests/scipy_tests/ndimage_tests|33626| 1 failed, 847 passed, 32775 skipped, 3 deselected in 101.43s (0:01:41) 
tests/cupyx_tests/scipy_tests/signal_tests|9307| short test summary info 
tests/cupyx_tests/scipy_tests/sparse_tests|24205| short test summary info 
tests/cupyx_tests/scipy_tests/spatial_tests|536| 536 skipped in 2.27s 
tests/cupyx_tests/scipy_tests/special_tests|483| 474 passed, 9 skipped, 34 warnings in 824.20s (0:13:44) 
tests/cupyx_tests/scipy_tests/stats_tests|201| 201 passed in 70.58s (0:01:10) 
tests/cupyx_tests/scipy_tests/test_get_array_module.py|2| 2 passed in 2.48s 
tests/cupyx_tests/test_cudnn.py|1829| 1829 skipped in 3.78s 
tests/cupyx_tests/test_cupyx.py|2| 2 passed in 0.99s 
tests/cupyx_tests/test_cusolver.py|648| 32 failed, 616 skipped in 5.47s 
tests/cupyx_tests/test_cusparse.py|1326| 803 failed, 14 passed, 146 skipped, 363 xfailed in 215.53s (0:03:35) 
tests/cupyx_tests/test_cutensor.py|79| 79 skipped in 1.14s 
tests/cupyx_tests/test_lapack.py|70| 33 failed, 8 passed, 29 skipped in 6.04s 
tests/cupyx_tests/test_optimize.py|9| 9 skipped in 1.08s 
tests/cupyx_tests/test_pinned_array.py|76| 72 passed, 4 deselected in 3.56s 
tests/cupyx_tests/test_rsqrt.py|1 item| 1 passed in 4.71s 
tests/cupyx_tests/test_runtime.py|2| 1 passed, 1 skipped in 2.10s 
tests/cupyx_tests/test_time.py|9| 1 failed, 8 passed in 2.62s 
tests/cupyx_tests/tools_tests|21| 6 passed, 15 deselected in 5.62s 
tests/example_tests|12| 5 failed, 2 passed, 5 skipped in 15.12s 
tests/install_tests|12| 9 passed, 3 skipped in 1.29s 

@lcskrishna

**Review by AdrianAbeyta — COMMENTED** [review]
Please post a log for rocm6.2_internal_testing branch before this PR's changes for UT failure comparison.

File: cupy_backends/stub/cupy_nvtx.h
Comment (AdrianAbeyta) [inline cupy_backends/stub/cupy_nvtx.h:1]: Should this be an empty file?

Comment (shbiswas834) [reply cupy_backends/stub/cupy_nvtx.h:1]: No it wasn't supposed to be there. I will remove it

**Comment by shbiswas834** [comment]
> Please post a log for rocm6.2_internal_testing branch before this PR's changes for UT failure comparison.

@AdrianAbeyta 

I looked through the logs of my build and these were the most common errors: [log file](https://amdcloud-my.sharepoint.com/:t:/g/personal/shbiswas_amd_com/ERAZZlwp5eZCrzP97XY3eNgBykiwRwnGr-PpSs8eDkQJhQ?e=bBUXf2)

[Cupy_ut_before_nvtx_logs](https://amdcloud-my.sharepoint.com/:u:/g/personal/shbiswas_amd_com/EaLXEGvI_S9HqhrNE0VMZ5gBBNam6Nsz5H8DM49rpTldrQ?e=aehRjs)

---------------------- TEST SUMMARY ------------------
tests/cupy_tests/array_api_tests|0| 8 skipped in 0.60s
tests/cupy_tests/binary_tests|15| 15 passed in 2.71s
tests/cupy_tests/core_tests|2872| short test summary info
tests/cupy_tests/creation_tests|506| 2 failed, 406 passed, 93 skipped, 5 deselected in 329.64s (0:05:29)
tests/cupy_tests/cuda_tests|268| 9 failed, 158 passed, 96 skipped, 6 deselected in 7.62s
tests/cupy_tests/fft_tests|2711| 437 failed, 1397 passed, 876 skipped, 1 deselected in 338.31s (0:05:38)
tests/cupy_tests/functional_tests|73| 73 passed, 1 warning in 794.72s (0:13:14)
tests/cupy_tests/indexing_tests|406| 405 passed, 1 deselected in 446.34s (0:07:26)
tests/cupy_tests/io_tests|42| 42 passed in 2.70s
tests/cupy_tests/lib_tests|576| 121 failed, 413 passed, 42 xfailed, 12 warnings in 251.57s (0:04:11)
tests/cupy_tests/linalg_tests|1395| 228 failed, 1151 passed, 16 skipped, 29 warnings in 444.76s (0:07:24)
tests/cupy_tests/logic_tests|824| 824 passed, 2 warnings in 280.41s (0:04:40)
tests/cupy_tests/manipulation_tests|585| short test summary info
tests/cupy_tests/math_tests|10192| test session starts
tests/cupy_tests/misc_tests|31| 31 passed in 14.11s
tests/cupy_tests/padding_tests|75| 75 passed in 14.91s
tests/cupy_tests/polynomial_tests|55| 55 passed in 43.71s
tests/cupy_tests/prof_tests|8| 8 passed in 1.01s
tests/cupy_tests/random_tests|1977| short test summary info
tests/cupy_tests/sorting_tests|499| short test summary info
tests/cupy_tests/statistics_tests|733| 7 failed, 723 passed, 3 deselected, 83 warnings in 736.93s (0:12:16)
tests/cupy_tests/test_cublas.py|2700| 2044 failed, 652 passed, 4 xfailed in 254.32s (0:04:14)
tests/cupy_tests/testing_tests|293| 29 failed, 231 passed, 23 skipped, 10 xfailed in 20.67s
tests/cupy_tests/test_init.py|18| 17 passed, 1 skipped in 4.91s
tests/cupy_tests/test_ndim.py|12| 12 passed in 2.47s
tests/cupy_tests/test_numpy_interop.py|11| 2 failed, 8 passed, 1 xfailed in 2.43s
tests/cupy_tests/test_type_routines.py|34| 34 passed in 1.89s
tests/cupy_tests/test_typing.py|1 item| 1 passed in 1.00s
tests/cupyx_tests/distributed_tests|59| 43 failed, 16 passed in 116.87s (0:01:56)
tests/cupyx_tests/fallback_mode_tests|107| 1 failed, 105 passed, 1 skipped in 8.39s
tests/cupyx_tests/jit_tests|172| 135 passed, 19 skipped, 18 xfailed, 22 warnings in 732.25s (0:12:12)
tests/cupyx_tests/linalg_tests|24| 20 failed, 5 skipped in 4.12s
tests/cupyx_tests/profiler_tests|19| 3 failed, 16 passed in 2.97s
tests/cupyx_tests/scipy_tests/fftpack_tests|901| 901 passed in 126.76s (0:02:06)
tests/cupyx_tests/scipy_tests/fft_tests|12237| 11657 passed, 420 skipped, 160 xpassed in 463.66s (0:07:43)
tests/cupyx_tests/scipy_tests/interpolate_tests|592| 223 failed, 248 passed, 120 skipped, 1 deselected in 85.84s (0:01:25)
tests/cupyx_tests/scipy_tests/linalg_tests|468| test session starts
tests/cupyx_tests/scipy_tests/ndimage_tests|33626| 1 failed, 847 passed, 32775 skipped, 3 deselected in 101.43s (0:01:41)
tests/cupyx_tests/scipy_tests/signal_tests|9307| short test summary info
tests/cupyx_tests/scipy_tests/sparse_tests|24205| short test summary info
tests/cupyx_tests/scipy_tests/spatial_tests|536| 536 skipped in 2.27s
tests/cupyx_tests/scipy_tests/special_tests|483| 474 passed, 9 skipped, 34 warnings in 824.20s (0:13:44)
tests/cupyx_tests/scipy_tests/stats_tests|201| 201 passed in 70.58s (0:01:10)
tests/cupyx_tests/scipy_tests/test_get_array_module.py|2| 2 passed in 2.48s
tests/cupyx_tests/test_cudnn.py|1829| 1829 skipped in 3.78s
tests/cupyx_tests/test_cupyx.py|2| 2 passed in 0.99s
tests/cupyx_tests/test_cusolver.py|648| 32 failed, 616 skipped in 5.47s
tests/cupyx_tests/test_cusparse.py|1326| 803 failed, 14 passed, 146 skipped, 363 xfailed in 215.53s (0:03:35)
tests/cupyx_tests/test_cutensor.py|79| 79 skipped in 1.14s
tests/cupyx_tests/test_lapack.py|70| 33 failed, 8 passed, 29 skipped in 6.04s
tests/cupyx_tests/test_optimize.py|9| 9 skipped in 1.08s
tests/cupyx_tests/test_pinned_array.py|76| 72 passed, 4 deselected in 3.56s
tests/cupyx_tests/test_rsqrt.py|1 item| 1 passed in 4.71s
tests/cupyx_tests/test_runtime.py|2| 1 passed, 1 skipped in 2.10s
tests/cupyx_tests/test_time.py|9| 1 failed, 8 passed in 2.62s
tests/cupyx_tests/tools_tests|21| 6 passed, 15 deselected in 5.62s
tests/example_tests|12| 5 failed, 2 passed, 5 skipped in 15.12s
tests/install_tests|12| 9 passed, 3 skipped in 1.29s

**Comment by shbiswas834** [comment]
Current NVTX Stub removal PR UT logs: [Logs](https://amdcloud-my.sharepoint.com/:u:/g/personal/shbiswas_amd_com/EebKorlqz6RFlrMMmlOIT8cBrE4bWSr-Db4GksnWxGsSgg?e=3QFFET)

Rocm 6.2 internal testing UT logs: [Logs](https://amdcloud-my.sharepoint.com/:u:/g/personal/shbiswas_amd_com/EaLXEGvI_S9HqhrNE0VMZ5gB_8H6D3RVHoBzulQHaaxWyg?e=VGUlu3)


UT logs before NVTX stub removal ---- After
image
image

File: cupy_backends/hip/cupy_roctx.h [hunk 1]
```diff
@@ -2,30 +2,77 @@
...
+    int32_t messageType;
+    nvtxMessageValue_t message;
+} nvtxEventAttributes_v1;
+
+typedef nvtxEventAttributes_v1 nvtxEventAttributes_t;
+
+// Fallback functions for NVTX using ROCtx
 
 void nvtxMarkA(const char* message) {
     roctxMarkA(message);
```
Comment (pnunna93) [inline cupy_backends/hip/cupy_roctx.h:52]: Please remove these wrappers if mappings are added to hipify torch

**Review by pnunna93 — CHANGES_REQUESTED** [review]
@shbiswas834 , please remove the wrappers which map nvtx to roctx directly as I mentioned in the comment. They need to be mapped in hipify torch if not present already. 

The file permissions seem to have changed, please revert them to original permissions.

**Comment by pnunna93** [comment]
Closing this PR as hipify torch flow is discarded.

**Pull request closed without merging**